### PR TITLE
VIMC-4580: Make orderly_cleanup chattier and sanitise report name

### PR DESCRIPTION
**Pull request by r-ash**
* This will now print out how many draft reports, csv files and rds files are going to be deleted
* And sanitise report name before running cleanup (i.e. strip / and strip preceeding `src`)

I think this should address #265

### CODE REVIEW

File: R/util.R [hunk 1]
```diff
@@ -925,3 +925,13 @@ lock_bindings <- function(syms, env) {
   }
   invisible(NULL)
 }
+
+clean_report_name <- function(name) {
```
Comment (richfitz) [inline R/util.R:929]: This duplicates logic in the increasingly misnamed `orderly:::orderly_develop_location`; can we use that instead?

File: R/cleanup.R [hunk 2]
```diff
@@ -85,12 +92,18 @@ orderly_cleanup_data <- function(config) {
 
   csv <- orderly_db("csv", config, FALSE)
   drop_csv <- setdiff(csv$list(), used)
+  msg <- sprintf("Found %s csv %s", length(drop_csv),
```
Comment (richfitz) [inline R/cleanup.R:95]: oh man totally forgot about all this old stuff. We should tidy this up at some point...

**Review by richfitz — CHANGES_REQUESTED** [review]

File: R/util.R (hunk 1 above)
Comment (r-ash) [reply R/util.R:929]: This PR pulls that file name cleaning logic out of `orderly_develop_location` into this new function. Then `orderly_develop_location` calls `clean_report_name`. Or you think `orderly_ cleanup` should call `orderly_develop_location`?

Comment (richfitz) [reply R/util.R:929]: Sorry, my mistake - you're right of course

**Review by richfitz — APPROVED** [review]

**Pull request merged**